### PR TITLE
Updated to PVWatts5 methodology in the pvwatts orientation estimation function system.infer_orientation_fit_pvwatts()

### LINKED ISSUE #147: Consider aligning `pvanalytics.system.infer_orientation_fit_pvwatts` with PVWatts better
`infer_orientation_fit_pvwatts` currently uses a basic power model sort of but not entirely resembling PVWatts ([v5](https://pvwatts.nrel.gov/downloads/pvwattsv5.pdf)?).  Some differences include:
- Assuming isotropic diffuse irradiance instead of the Perez transposition model
- Using the Sandia cell temperature model instead of Fuentes
- No IAM model
- Different default temperature coefficient (-0.004 vs -0.0047)
- No "pvwatts losses"

Question to discuss: should these differences be resolved so that it implements the full PVWatts model?  Using `pvlib.modelchain.ModelChain.with_pvwatts` would get it close (there is still a temperature model difference, as of pvlib 0.9.1).  

Either way, as discussed in https://github.com/pvlib/pvanalytics/pull/77#discussion_r511010737, the docstring `Notes` section could use a small revision.

### PR DESCRIPTION
**Pull request by kperrynrel**
Update the PVWatts calculations to better align with PVWatts 5 methodology. 

- [x] Closes #147 
- [x] Added tests to cover all new or modified code.
- [x] Clearly documented all new API functions with [PEP257](https://www.python.org/dev/peps/pep-0257/) and [numpydoc](https://numpydoc.readthedocs.io/en/latest/format.html) compliant docstrings.
- [x] Added new API functions to `docs/api.rst`.
- [x] Non-API functions clearly documented with docstrings or comments as necessary.
- [x] Adds description and name entries in the appropriate "what's new" file 
      in [`docs/whatsnew`](https://github.com/pvlib/pvanalytics/tree/master/docs/whatsnew) 
      for all changes. Includes link to the GitHub Issue with `` :issue:`num` `` 
      or this Pull Request with `` :pull:`num` ``. Includes contributor name 
      and/or GitHub username (link with `` :ghuser:`user` ``).
- [x] Pull request is nearly complete and ready for detailed review.
- [x] Maintainer: Appropriate GitHub Labels and Milestone are assigned to the Pull Request and linked Issue.

### CODE REVIEW

File: pvanalytics/system.py [hunk 1]
```diff
@@ -493,7 +518,9 @@ def infer_orientation_fit_pvwatts(power_ac, ghi, dhi, dni,
                                   solar_zenith, solar_azimuth,
                                   temperature=25, wind_speed=0,
                                   temperature_coefficient=-0.004,
```
Comment (kanderso-nrel) [inline pvanalytics/system.py:520]:
```suggestion
                                  temperature_coefficient=-0.0047,
```

File: pvanalytics/system.py [hunk 2]
```diff
@@ -461,25 +471,40 @@ def _power_residuals_from_clearsky(system_params,
...
+        dni_extra=dni_extra,
+        airmass=relative_airmass,
+        model='perez'
     )
+    # Get AOI
+    aoi = pvlib.irradiance.aoi(tilt, azimuth,
+                               solar_zenith,
+                               solar_azimuth)
+    # Run IAM model
+    iam = pvlib.iam.physical(aoi)
```
Comment (kanderso-nrel) [inline pvanalytics/system.py:489]:
```suggestion
    iam = pvlib.iam.physical(aoi, n=1.5)
```

File: pvanalytics/system.py [hunk 3]
```diff
@@ -493,7 +518,9 @@ def infer_orientation_fit_pvwatts(power_ac, ghi, dhi, dni,
                                   solar_zenith, solar_azimuth,
                                   temperature=25, wind_speed=0,
                                   temperature_coefficient=-0.004,
-                                  temperature_model_parameters=None):
+                                  temperature_model_parameters=None,
+                                  azimuth_min=0,
+                                  azimuth_max=360):
     """Get the tilt and azimuth that give PVWatts output that most closely
```
Comment (kanderso-nrel) [inline pvanalytics/system.py:524]:
```suggestion
    """Get the tilt and azimuth that give PVWatts v5 output that most closely
```

File: pvanalytics/system.py [hunk 4]
```diff
@@ -547,6 +574,10 @@ def infer_orientation_fit_pvwatts(power_ac, ghi, dhi, dni,
         ``pvlib.temperature.TEMPERATURE_MODEL_PARAMETERS['sapm'][
         'open_rack_glass_glass']`` is used. See
         :py:func:`pvlib.temperature.sapm_cell` for more information.
+    azimuth_min: Float, default 0
+        Minimum possible azimuth (bounds) for the least squares search problem.
+    azimuth_max: Float, default 360
+        Maximum possible azimuth (bounds) for the least squares search problem.
```
Comment (kanderso-nrel) [inline pvanalytics/system.py:580]: If we're exposing these as optional parameters, should the other bounds (tilt's 0 to 90, e.g.) be exposed as well?

Also, `[degrees]` at the ends of the descriptions

File: docs/whatsnew/0.1.4.rst [hunk 5]
```diff
@@ -5,6 +5,13 @@
 
 Enhancements
 ~~~~~~~~~~~~
+* Updated function :py:func:`~pvanalytics.system.infer_orientation_fit_pvwatts`
+  to more closely align with the PVWatts v5 methodology. This includes incorporating
```
Comment (kanderso-nrel) [inline docs/whatsnew/0.1.4.rst:9]: I'm torn on whether this counts as an `Enhancement` (which can go into 0.1.4) or a breaking change that requires 0.2.0. Curious what others think here.

**Review by kanderso-nrel — COMMENTED** [review]
The `Notes` in the residual function's docstring is no longer relevant; I'd just delete it.  I suggest also adding something like this to `infer_orientation_fit_pvwatts`: 

```
Notes
-----
To prevent significant slowdown, this function uses the SAPM
thermal model (:py:func:`~pvlib.temperature.sapm_cell`) instead of
the model specified in the PVWatts v5 reference [1]_
(:py:func:`~pvlib.temperature.fuentes`).

References
----------
.. [1] Aron Dobos, "PVWatts Version 5 Manual", NREL/TP-6A20-62641 (2014).
       :doi:`10.2172/1158421`
```

**Comment by kperrynrel** [comment]
I was able to get some results on how this new enhancement changes the estimations, by comparing previous results (for the existing function) to new PVWatts 5 results for the 23 system dataset we've been testing data out on (196 data streams total). When I take the absolute difference for (old estimate-new estimate), I generate the following histograms:

image
image

Azimuth variation, abs(old-new), median: 0.88, mean: 3.98
Tilt variation, abs(old-new), median: 0.88, mean: 1.29

Variation is pretty minor. To determine if the PVwatts 5 function is more accurate, I checked if the results for the new PVWatts5 function improved upon the previous function results:

Old model azimuth estimation MAE (compared to ground truth): 10.23
New model azimuth estimation MAE (compared to ground truth): 8.064

Old model tilt estimation MAE (compared to ground truth): 2.185
New model tilt estimation MAE (compared to ground truth): 1.60

The above results are promising, as it means that the new PVwatts5 function is more accurately estimating azimuth and tilt than the old function.

File: docs/whatsnew/0.1.4.rst (hunk 5 above)
Comment (cwhanse) [reply docs/whatsnew/0.1.4.rst:9]: I think it's a breaking change that merits 0.2.0.

**Review by kanderso-nrel — APPROVED** [review]
One last minor change to better match PVWatts v5 (@kperrynrel I doubt this will materially affect your validation results), otherwise LGTM.  

For transparency: the plan is to merge this and tag a beta release for 0.2.0.  We thought it better to hedge with a beta release for now in case we end up wanting to make further changes to this function in the course of @kperrynrel's validation for PVSC.

**Pull request merged**